packages/beacon-node/src/api/impl/beacon/state/utils.ts

```diff
@@ -187,7 +188,7 @@ export function getStateValidatorIndex(
 
   // typeof id === Uint8Array
   const validatorIndex = pubkey2index.get(id);
-  if (validatorIndex === undefined) {
+  if (validatorIndex === null) {
```

Comment:
this is generated by napi-rs so I just follow the interface there https://github.com/ChainSafe/pubkey-index-map/blob/b2f2ba42850aaf131559c8790174260e996b475a/index.d.ts#L9

we can have a proxy map to return `undefined` instead of `null` to have less code change here, but I think doing that would make it a bit hard to understand in the future for the maintainer

Reply:
Ok, lets just keep it as is, returning `number | null`

Reply:
I also tend to like `null` to differentiate "intended non-result" vs "no value set" which could be a lot of other things